LEAF_Request_Portal/admin/templates/mod_workflow.tpl

```diff
@@ -2177,85 +2181,89 @@
+                workflows[workflowID]['description'] = title;
+                old_steps[-1] = -1;
 
-            for(let i in steps) {
-                // add step, if successful update that step
-                addStep(workflowID, steps[i].stepTitle, function(stepID) {
+                for(let i in steps) {
+                    // add step, if successful update that step
+                    addStep(workflowID, steps[i].stepTitle, function(stepID) {
 
-                    if (isNaN(stepID)) {
-                        console.log(stepID);
-                    } else {
-                        old_steps[steps[i].stepID] = stepID;
-                        updatePosition(workflowID, stepID, steps[i].posX, steps[i].posY);
+                        if (isNaN(stepID)) {
+                            console.log(stepID);
+                        } else {
+                            old_steps[steps[i].stepID] = stepID;
+                            updatePosition(workflowID, stepID, steps[i].posX, steps[i].posY);
 
-                        updateTitle(steps[i].stepTitle, stepID, function(res) {
-                            // Alls well that ends well.
-                        });
+                            updateTitle(steps[i].stepTitle, stepID, function(res) {
+                                // Alls well that ends well.
+                            });
 
-                        if (steps[i].stepData != null) {
-                            let auto = JSON.parse(steps[i].stepData);
+                            if (steps[i].stepData != null) {
+                                let auto = JSON.parse(steps[i].stepData);
 
-                            let seriesData = {
-                                AutomatedEmailReminders: {
-                                    'Automate Email Group': auto.AutomatedEmailReminders.AutomateEmailGroup,
-                                    'Days Selected': auto.AutomatedEmailReminders.DaysSelected,
-                                    'Date Selected': auto.AutomatedEmailReminders?.DateSelected || '',
-                                    'Additional Days Selected': auto.AutomatedEmailReminders.AdditionalDaysSelected,
-                                }
-                            };
+                                let seriesData = {
+                                    AutomatedEmailReminders: {
+                                        'Automate Email Group': auto.AutomatedEmailReminders.AutomateEmailGroup,
+                                        'Days Selected': auto.AutomatedEmailReminders.DaysSelected,
+                                        'Date Selected': auto.AutomatedEmailReminders?.DateSelected || '',
+                                        'Additional Days Selected': auto.AutomatedEmailReminders.AdditionalDaysSelected,
+                                    }
+                                };
+
+                                updateStepData(seriesData, stepID, function (res) {
```

Comment:
Why is an empty function being passed in?

Reply:
My only change here was adding the if/else wrapper to prevent duplication of a built-in workflow.  Everything else in this function is from the duplicate workflow feature last summer.  It's easier to see if the master and branch files are side by side - not sure why just wrapping it made the diff extra crazy.

Reply:
If there is something fundamentally wrong with the workflow duplication feature, then I think it should be handled in a ticket specific to that.  This ticket was created to address an issue that makes it possible to unintentionally modify a built in workflow.  The only change for this function was a wrapper to prevent duplication of a built-in.  I guess since it added a tab to everything inside, it make it appear that there were a number of changes here (I had to look at the files side by side as a sanity check myself).